### PR TITLE
chore(nextjs): Refactor Next.js integration using the isomorphic @clerk/backend

### LINKED ISSUE #585: Bug: atob invalid character exception
- [x] Review the documentation: https://docs.clerk.dev/
- [x] Search for existing issues: https://github.com/clerkinc/javascript/issues
- [x] Go through package changelog files.
- [x] Provide the Frontend API key from your application dashboard.

## Package + Version

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] other:

### Version:

```
4.5.2 also tested on latest 4.6.5
```

### Browser/OS

If applicable e.g. `Chrome latest`

## Description

When using a custom session claims with `profile_image` the Clerk `getAuth` function breaking on invalid character using `atob` function. Getting the session token and unwrapping the image url is set to `"profile_image": "https://www.gravatar.com/avatar?d=mp"` and we think the `=` is probably breaking it. Here is the stack trace

Here is our custom claims object

```
{
	"email": "{{user.primary_email_address}}",
	"user_id": "{{user.id}}",
	"last_name": "{{user.last_name}}",
	"first_name": "{{user.first_name}}",
	"email_verfied": "{{user.email_verified}}"
}
```

Stack trace error on login

```
error - DOMException [InvalidCharacterError]: Invalid character
    at atob (node:buffer:1247:13)
    at parseRequest (/Users/arashsoheili/Code/caredge-app/node_modules/@clerk/nextjs/dist/server/utils/getAuth.js:49:38)
    at /Users/arashsoheili/Code/caredge-app/node_modules/@clerk/nextjs/dist/server/utils/getAuth.js:24:61
    at eval (webpack-internal:///(api)/./src/api/common/handler/method.ts:67:83)
    at resolver (webpack-internal:///(api)/./src/api/common/handler/index.ts:109:19)
    at Object.apiResolver (/Users/arashsoheili/Code/caredge-app/node_modules/next/server/api-utils/node.ts:522:11)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async DevServer.runApi (/Users/arashsoheili/Code/caredge-app/node_modules/next/server/next-server.ts:789:5) {
  page: '/api/auth/vehicles-owned'
}
```

Have tested removing the profile_image from custom session and we then have no issues.

The `atob` function is deprecate. Here is the docs from nodejs

```
This function is only provided for compatibility with legacy web platform APIs and should never be used in new code, because they use strings to represent binary data and predate the introduction of typed arrays in JavaScript. For code running using Node.js APIs, converting between base64-encoded strings and binary data should be performed using Buffer.from(str, 'base64') and buf.toString('base64').
```

### PR DESCRIPTION
**Pull request by SokratisVidros**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

- [x] Leverage the new isomorphic `@clerk/backend` package and reduce the complexity of our Next.js integration.
- [x] Use the new terminology of RequestState and AuthObject in @clerk/nextjs

For now we updated only the Next.js > 12.2 integration paths assuming that customers use Next.js Middleware. Updating the legacy paths might follow.

This refactoring also fixes #585

### CODE REVIEW

File: packages/nextjs/src/server/getAuth.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * To enable Clerk SSR support, include this object to the `props`
+ * returned from `getServerSideProps`. This will automatically make the auth state available to
+ * the Clerk components and hooks during SSR, the hydration phase and CSR.
+ * @example
+ * import { getAuth } from '@clerk/nextjs/server';
+ *
+ * export const getServerSideProps = ({ req }) => {
+ *   const { authServerSideProps } = getAuth(req);
+ *   const myData = getMyData();
```
Comment (nikosdouvlis) [inline packages/nextjs/src/server/getAuth.ts:54]: This comment is tale as it refers to a previous version of this api.

**Review by nikosdouvlis — APPROVED** [review]
This looks good. I'm a bit unsure about the release strategy here. Could you share more details? Are we hiding the refactor behind a temporary experimental module? Or will we ship a v5 major version?

**Comment by SokratisVidros** [comment]
@nikosdouvlis Is this ready for final review?

**Comment by nikosdouvlis** [comment]
As discussed, closing this one in favor of https://github.com/clerkinc/javascript/pull/630

**Pull request closed without merging**